### PR TITLE
feat: add Scheduler options

### PR DESCRIPTION
**Pull request by tychoish**
This is just an example, to see if you all would be interested in
adopting a feature like this. If this is viable, I would want to write
tests and better documentation. 

Thanks for your time and consideration!

### CODE REVIEW

**Comment by reugn** [comment]
Hi @tychoish, thanks for your interest in the project.
This feature can prevent jobs in the queue from running in time. Also, what would be a real use case to block in the scheduler?

**Comment by tychoish** [comment]
We're running into an issue where jobs that run longer than their interval end up piling up. I think we'd like to be able to block to prevent the pileup and ensure only one copy of the job is running at once. I think your critique is fair, but I think keeping it as opt-in (potentially with even better documentation) removes a lot of the risk.

**Comment by tychoish** [comment]
I added a second option, which I think makes this even more tenable, that dispatches all jobs to a worker pool which may be more consistent with the kind of semantics that users would want. 

The default behavior remains unchanged.

**Comment by tychoish** [comment]
I've added more tests for the worker pool, which should cover most of the gap. 

It definitely doesn't pass the race detector at the moment because the implementation of FunctionJob isn't thread safe, but I think that's always been the case.

File: quartz/scheduler_test.go [hunk 1]
```diff
@@ -65,6 +66,101 @@ func TestScheduler(t *testing.T) {
...
+			case "WorkerSmall", "WorkerLarge":
+			WORKERS:
+				for iters := 0; iters < 100; iters++ {
+					select {
+					case <-ctx.Done():
+						break WORKERS
+					case <-ticker.C:
+						num := atomic.LoadInt64(&n)
+						if num > int64(opts.WorkerLimit) {
+							t.Errorf("on iter %d n %d was more than limt %d",
```
Comment (reugn) [inline quartz/scheduler_test.go:151]:
```suggestion
							t.Errorf("on iter %d n %d was more than limit %d",
```

**Review by reugn — APPROVED** [review]
LGTM

**Pull request merged**